settings.gradle.kts

```diff
@@ -1,15 +1,75 @@
+    repositories {
+        mavenCentral()
+        gradlePluginPortal()
     }
 }
+
+@Suppress("UnstableApiUsage")
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
+
+    repositories {
+        mavenCentral()
+
+        // Declare the Node.js & Yarn download repositories
+        exclusiveContent {
+            forRepository {
+                ivy("https://nodejs.org/dist/") {
+                    name = "Node Distributions at $url"
+                    patternLayout { artifact("v[revision]/[artifact](-v[revision]-[classifier]).[ext]") }
+                    metadataSources { artifact() }
+                    content { includeModule("org.nodejs", "node") }
+                }
+            }
+            filter { includeGroup("org.nodejs") }
+        }
+
+        exclusiveContent {
+            forRepository {
+                ivy("https://github.com/yarnpkg/yarn/releases/download") {
+                    name = "Yarn Distributions at $url"
+                    patternLayout { artifact("v[revision]/[artifact](-v[revision]).[ext]") }
+                    metadataSources { artifact() }
+                    content { includeModule("com.yarnpkg", "yarn") }
+                }
+            }
+            filter { includeGroup("com.yarnpkg") }
+        }
+
+        // workaround for https://youtrack.jetbrains.com/issue/KT-51379
+        exclusiveContent {
+            forRepository {
+                ivy("https://download.jetbrains.com/kotlin/native/builds") {
+                    name = "Kotlin Native"
+                    patternLayout {
+
+                        // example download URLs:
+                        // https://download.jetbrains.com/kotlin/native/builds/releases/1.7.20/linux-x86_64/kotlin-native-prebuilt-linux-x86_64-1.7.20.tar.gz
+                        // https://download.jetbrains.com/kotlin/native/builds/releases/1.7.20/windows-x86_64/kotlin-native-prebuilt-windows-x86_64-1.7.20.zip
+                        // https://download.jetbrains.com/kotlin/native/builds/releases/1.7.20/macos-x86_64/kotlin-native-prebuilt-macos-x86_64-1.7.20.tar.gz
+                        listOf(
+                            "macos-x86_64",
+                            "macos-aarch64",
+                            "osx-x86_64",
+                            "osx-aarch64",
+                            "linux-x86_64",
+                            "windows-x86_64",
+                        ).forEach { os ->
+                            listOf("dev", "releases").forEach { stage ->
+                                artifact("$stage/[revision]/$os/[artifact]-[revision].[ext]")
+                            }
+                        }
+                    }
+                    metadataSources { artifact() }
+                }
+            }
+            filter { includeModuleByRegex(".*", ".*kotlin-native-prebuilt.*") }
+        }
+    }
+}
+
+enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")
```

Comment:
You can read more about the specifics of the [feature lifecycle in the Gradle docs](https://docs.gradle.org/current/userguide/feature_lifecycle.html).

* [`STABLE_CONFIGURATION_CACHE`](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:stable) just adds some additional warnings if there's a problem with [Configuration Cache](https://docs.gradle.org/current/userguide/configuration_cache.html). My motivation for enabling this is because at some point Config Cache is going to be enabled by default (I think), so it's nice to get the warnings now so that the problems can be fixed sooner.

* [`TYPESAFE_PROJECT_ACCESSORS`](https://docs.gradle.org/7.0/release-notes.html#type-safe-project-accessors) generates some handy Kotlin DSL utils for adding cross project dependencies. When combined with the Version Catalog accessors it helps make the build config files look really slick and easy to read. There are a [few issues](https://github.com/gradle/gradle/labels/in%3Aproject-accessors), mostly edge-cases related when subprojects have unexpected characters or with IDE support, but I've never noticed an issue in my projects.

   This flag is actually not useful for Fleks because there's only one subproject 🙃. So there's no harm in keeping it, but no harm in removing it.


These options are safe, I use them in all my projects, but they're not mandatory and if you want to remove them I won't argue :)

Reply:
Thank you for the explanation and links to the documentations. Let's keep it then! Makes sense to me :)